src/parser.c

```diff
@@ -46,25 +46,24 @@ void init_parser(FILE *fp)
     next_token(fp);
 }
 
-stringArray *parse_to_arr(FILE *fp)
+char **parse_to_arr(FILE *fp)
 {
 	init_parser(fp);
-	stringArray *arr;
-	arr = malloc(sizeof(stringArray));
-	arr->array = malloc(sizeof(char*)*MAXWORDNUM);
-    assert(arr != NULL);
+	char **strarr;
+	strarray = malloc(sizeof(char*)*MAXWORDNUM);
```

Comment:
Where is `strarray` coming from? Is line 52 a typo? How are you then asserting `strarr != NULL` on line 54 without having `malloc`ed it? I'm so confused. If `strarr` and `strarray` are meant to be the same, please fix the typo. I'm mystified at how the tests are passing. If they're meant to be different variables, please avoid naming variables so similarly.

Reply:
changed strarray to strarr